src/GraphQL/Builders/FieldBuilder.cs

```diff
@@ -107,10 +108,39 @@ protected FieldBuilder(FieldType fieldType)
         /// <param name="type">The graph type of the field.</param>
         public virtual FieldBuilder<TSourceType, TReturnType> Type(IGraphType type)
         {
+            FieldType.Type = null;
             FieldType.ResolvedType = type;
             return this;
         }
 
+        /// <summary>
+        /// Sets the graph type of the field.
+        /// </summary>
+        public virtual FieldBuilder<TSourceType, TReturnType> Type<TGraphType>()
+            where TGraphType : IGraphType
+        {
+            FieldType.Type = typeof(TGraphType);
+            FieldType.ResolvedType = null;
+            return this;
+        }
+
+        /// <summary>
+        /// For input object fields, sets the mapping of the field to the source type's member.
+        /// </summary>
+        [Obsolete("Please configure the field mapping by providing an expression as an argument to the 'Field' method.")]
+        public virtual FieldBuilder<TSourceType, T> Map<T>(Expression<Func<TSourceType, T>> expression)
```

Comment:
This method should work perfectly with all existing code, since ToObject is already aware of the `ORIGINAL_EXPRESSION_PROPERTY_NAME` metadata.  But I don't think we should add this method; rather people should use the `Field("testing", x => x.Sample)` when defining their field.  Thoughts?

Reply:
> rather people should use the Field("testing", x => x.Sample)

This syntax is less discoverable. I wasn't aware of it before you showed it to me when working on analyzers.

Reply:
Also, I don't know how others define their schema, but we always start with `Field<>(name)...`. With the syntax above, the input field definition is not consistent with other object fields.

Reply:
The `ResolverAnalyzer` (GQL005) will remove the `Resolve` from the input fields. I think it is worth adding a fix to convert it to `Map` instead. So whoever has not upgraded to [7.4](https://github.com/graphql-dotnet/graphql-dotnet/pull/3574) yet will get the desired behavior.

Reply:
The syntax below has been the primary way of adding fields to both input and output objects since 2.x.  I really don't think we should change it.

```cs
Field(x => x.Name); // add a field for the Name property, non-nullable, with an auto-mapped graph type
Field(x => x.Name, true); // similar but nullable
Field("FullName", x => x.Name); // similar but with "fullName" as the name, non-nullable
Field("FullName", x => x.Name, true); // similar but nullable
```

If this PR was merged with no changes, the equivalent code to the first line would be this:

```cs
Field<NonNullGraphType<GraphQLClrOutputTypeReference<string>>, string>("Name").Map(x => x.Name);
```

It is much desirable to allow users to easily create graph types such as this:

```cs
public class CustomerType : ObjectGraphType<Customer>
{
    public CustomerType()
    {
        Field(x => x.FirstName);
        Field(x => x.LastName);
        Field("Name", x => x.FullName);
        Field(x => x.Company, true);
    }
}
```

The code would look almost identical for an input type.

Reply:
We never used this overload. All our fields defined in a more verbose but explicit way
```c#
public class CustomerType : ObjectGraphType<Customer>
{
    public CustomerType()
    {
        Field<NonNullGraphType<StringGraphType>>("FirstName).Resolve(ctx => ctx.Source.FirstName);
        Field<NonNullGraphType<StringGraphType>>("LastName).Resolve(ctx => ctx.Source.LastName);
        Field<NonNullGraphType<StringGraphType>>("Name).Resolve(ctx => ctx.Source.FullName);
        Field<StringGraphType>("Company).Resolve(ctx => ctx.Source.Company);
    }
}
```

We do it mainly to decouple graph field names from the source property names. But I agree that the same can be achieved with a more compact form.
```c#
public class CustomerType : ObjectGraphType<Customer>
{
    public CustomerType()
    {
        Field("FirstName", x => x.FirstName);
        Field("LastName", x => x.LastName);
        Field("Name", x => x.FullName);
        Field("Company", x => x.Company, true);
    }
}
```

Maybe we should adopt this.